### PR TITLE
Package splitting and documentation updates

### LINKED ISSUE #15: JRPCServer comment is invalid
It refers `#sendRequest:` method but this method does not exist in this class.

### LINKED ISSUE #16: Update documentation related to server to introduce message processor concept

### LINKED ISSUE #17: Update documentation to introduce handler concept

### LINKED ISSUE #18: Create more packages to allow users to depend only on what they need
I think packages are missing for the server:
- `JRPC-Server-TCP`
- `JRPC-Server-HTTP`
- `JRPC-Server-Core` This package could be used if users only use the message-processor.

I would like to add support for web socket as transport layer in the future.

For the clients we will have similar packages:
- `JRPC-Client-TCP`
- `JRPC-Client-HTTP`

What do you think @gcotelli ?

### PR DESCRIPTION
**Pull request by gcotelli**
Fixes #15 
Fixes #16 
Fixes #17 
Fixes #18

@juliendelplanque please review the handlers documentation

### CODE REVIEW

**Comment by gcotelli** [comment]
@juliendelplanque I think the PR is ready for review. I don't get why the coverage decreased given I've just moved code from one package to other packages. Maybe it depends on the way Coveralls combines the coverage info from different packages?

File: docs/MessageProcessing.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Message Processing
+
+JSON RCP messages are processed by instances of `JRPCMessageProcessor` independently of the underlying transport used for receiving and responding the messages.
+
+The message processor has available a list of method handlers that will be searched when a incoming message is received, if a suitable handler is found then it gets executed with the parameters provided in the message.
```
Comment (juliendelplanque) [inline docs/MessageProcessing.md]:
```suggestion
The message processor has a list of method handlers available that will be searched when a incoming message is received. If a suitable handler is found (the `#methodName` is matching) then it gets executed with the parameters provided in the message.
```

File: docs/MessageProcessing.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+To configure a message processor use any of the following messages to add new handlers:
```
Comment (juliendelplanque) [inline docs/MessageProcessing.md]:
```suggestion
A message processor can be configured using the following messages:
```

File: docs/MessageProcessing.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+- `addHandlerNamed:block:` will add a new handler for a method that will evaluate the provided block
```
Comment (juliendelplanque) [inline docs/MessageProcessing.md]:
```suggestion
- `addHandlerNamed:block:`adds a new handler for a method that will evaluate the block provided as argument.
```

File: docs/MessageProcessing.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+- `addHandlersFromPragmasIn:` will search for `<jrpc>` pragmas declared in the methods of the parameter object, and will add a message handler if found.
```
Comment (juliendelplanque) [inline docs/MessageProcessing.md]:
```suggestion
- `addHandlersFromPragmasIn:` searches for `<jrpc>` pragmas declared in the methods of the object provided as argument, and adds a message handler if such pragma is found. When such handler is added, the method in which the pragma was found is sent to the object when a matching JRPC message is received.
```

File: docs/MessageProcessing.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+- `addHandler:` will add a new handler. Handlers must subclass `JRPCAbstractHandler`, this allows the end user to extend the available options if needed.
```
Comment (juliendelplanque) [inline docs/MessageProcessing.md]:
```suggestion
- `addHandler:` adds a new handler. Handlers must subclass `JRPCAbstractHandler`, this allows the end user to extend the available options if needed.
```

File: docs/MessageProcessing.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+The message processor main API is `handleJSON:` , any server implementing an specific transport protocol must obtain the json string and delegate it's processing by sending this message and will receive the json string to be used as response.
```
Comment (juliendelplanque) [inline docs/MessageProcessing.md]:
```suggestion
The message processor main API is `handleJSON:`, any server implementing a specific transport protocol must obtain the json string and delegate its processing by sending `handleJSON:` to the processor. `handleJSON:` method returns the JSON string to return to the client once the processing is completed.
```

File: docs/MessageProcessing.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+## Message handlers
+
+Message handlers are responsible of executing the code related to a method. All handlers are subclasses of `JRPCAbstractHandler` and need to provide a `methodName` and a way to execute this method with the provided arguments. There's two implementations already available `JRPCBlockHandler` and `JRPCPragmaHandler`.
```
Comment (juliendelplanque) [inline docs/MessageProcessing.md]:
```suggestion
Message handlers are responsible of executing the code related to a JRPC method. All handlers are subclasses of `JRPCAbstractHandler` and need to provide a `methodName` and a way to execute this method with the provided arguments. There's two implementations already available `JRPCBlockHandler` and `JRPCPragmaHandler`.
```

File: docs/MessageProcessing.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+- `JRPCBlockHandler` instances are created with a block closure that will we evaluated to compute the method. The parameter names are taken from the block.
```
Comment (juliendelplanque) [inline docs/MessageProcessing.md]:
```suggestion
- `JRPCBlockHandler` instances are created with a block closure that will be evaluated to compute the method. The parameter names are taken from the block parameters names.
```

**Review by juliendelplanque — CHANGES_REQUESTED** [review]

**Comment by gcotelli** [comment]
Applied code review suggestions

**Review by juliendelplanque — APPROVED** [review]

**Pull request merged**